src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -1010,5 +1013,48 @@ async def setup_venv_pex_process(request: VenvPexProcess) -> Process:
     )
 
 
+@dataclass(frozen=True)
+class PexDistributionInfo:
+    """Information about an individual distribution in a PEX file, as reported by
+    `PEX_TOOLS=1 repository info -v`."""
+
+    project_name: str
+    version: str
+    requires_python: str | None
+    requires_dists: list[str]
+
+
+class PexDistributionsInfo(Collection[PexDistributionInfo]):
```

Comment:
I don't like how similar these two names are, but couldn't come up with a better one after time boxing myself.

Reply:
Its really `ResolveInfo` if it's sill bugging you.